### PR TITLE
Highlight columns in Channel Comparison for quick grammar checking

### PR DESCRIPTION
**Pull request by TheoChevalier**
One of the missing pieces to fix issue #569. Menu would be another PR. I was thinking about it, and turns out that highlighting text is making it easier than copying values directly

I still need to figure out how to make the click events generic (e.g. not relying on IDs) so that we have a single call at the bottom of select_column.js that would work for, let's say, all tables with 'selectable' class.

Also need to figure out the appropriate UI (icon, button…)

### CODE REVIEW

File: web/js/select_column.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    } else {
+        cells.each(function(i, cell) {
+            var rangeObj = document.body.createTextRange();
+            rangeObj.moveToElementText(cell);
+            rangeObj.select();
+        });
+    }
+}
+
+$(document).ready(function() {
```
Comment (flodolo) [inline web/js/select_column.js]: You can drastically simplify this: assign a class `select_header` to the columns you want to be selectable (right now you let select also the key column, which is not useful), and use
```JS
$(document).ready(function() {
    $('.select_header').click(function() {
        var columnNumber = $(this).index();
        var tableID = $(this).closest('table').attr('id');
        SelectColumn(columnNumber, tableID);
    });
});
```

File: web/js/select_column.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    if (document.createRange) {
+        cells.each(function(i, cell) {
+            var rangeObj = document.createRange();
+            rangeObj.selectNodeContents(cell);
+            window.getSelection().addRange(rangeObj);
+        });
+
+    // Internet Explorer before version 9
```
Comment (flodolo) [inline web/js/select_column.js]: To which part of the code is this comment related?

File: web/js/select_column.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+function SelectColumn(index, tableId) {
+    var columnSelector = '#' + tableId + ' tbody > tr > td:nth-child(' + (index + 1) + ')';
+    var cells = $(columnSelector);
+
+    // clear existing selections
```
Comment (flodolo) [inline web/js/select_column.js]: nit: Clear

**Comment by TheoChevalier** [comment]
Added a basic UI

File: web/style/transvision.css [hunk 4]
```diff
@@ -296,6 +296,15 @@ th {
     padding: 0.5em;
 }
 
+th.select_header::after {
+    content: "Select";
+    font-weight: normal;
```
Comment (flodolo) [inline web/style/transvision.css]: I'm not a huge fan: 
- The existing label is not centered anymore.
- It's not available when table collapses (maybe that's OK).
- The cursor should change on hover.
- The text should probably say "Select column" for more clarity.

What about adding a real element in the HTML (e.g. a `<span>`) and use absolute positioning. We could also re-use the copy icon, and use "Select column" as alt/title

File: web/js/select_column.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+function SelectColumn(index, tableID) {
```
Comment (flodolo) [inline web/js/select_column.js]: Nit: selectColumn

File: web/style/transvision.css (hunk 4 above)
Comment (pascalchevrel) [inline web/style/transvision.css]: For the cursor, I suggest:
```css
cursor: cell;
```

**Comment by pascalchevrel** [comment]
the en-US column for the "New strings added in fr between Aurora and Beta" section has no select button, I think it should both for consistency with the first table and because I can imagine some people interested in copy/pasting this data easily

**Comment by TheoChevalier** [comment]
Thanks, updated

**Comment by flodolo** [comment]
It looks good, I wonder if we should we worry about the completely collapsed table?

**Comment by TheoChevalier** [comment]
IMO that’s a feature, not a bug (e.g. not getting clumsy UI when you don’t need it).
People using grammar checker on tablets, that would surprise me :)

**Comment by flodolo** [comment]
I guess that makes sense :smile:
